stl/inc/xspan

```diff
@@ -0,0 +1,238 @@
+
+    // [span.sub], span subviews
+    template <size_t _Count>
+    constexpr span<element_type, _Count> first() const noexcept {
+        if constexpr (_Extent != dynamic_extent) {
+            static_assert(_Count <= _Extent, "Count out of range in span::first()");
+        } else {
+#if _CONTAINER_DEBUG_LEVEL > 0
+            _STL_VERIFY(_Count <= size(), "Count out of range in span::first()");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        }
+        return {data(), _Count};
+    }
+
+    constexpr span<element_type, dynamic_extent> first(index_type _Count) const noexcept {
+#if _CONTAINER_DEBUG_LEVEL > 0
+        _STL_VERIFY(_Count <= size(), "Count out of range in span::first(Count)");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        return {data(), _Count};
+    }
+
+    template <size_t _Count>
+    constexpr span<element_type, _Count> last() const noexcept {
+        if constexpr (_Extent != dynamic_extent) {
+            static_assert(_Count <= _Extent, "Count out of range in span::last()");
+        } else {
+#if _CONTAINER_DEBUG_LEVEL > 0
+            _STL_VERIFY(_Count <= size(), "Count out of range in span::last()");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        }
+        return {data() + size() - _Count, _Count};
+    }
+
+    constexpr span<element_type, dynamic_extent> last(index_type _Count) const noexcept {
+#if _CONTAINER_DEBUG_LEVEL > 0
+        _STL_VERIFY(_Count <= size(), "Count out of range in span::first(Count)");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        return {data() + size() - _Count, _Count};
+    }
+
+    template <size_t _Offset, size_t _Count = dynamic_extent>
+    constexpr span<element_type, dynamic_extent> subspan() const noexcept {
+#if _CONTAINER_DEBUG_LEVEL > 0
+        _STL_VERIFY(_Offset <= size(), "Offset out of range in span::subspan()");
+        _STL_VERIFY(_Count == dynamic_extent || _Offset + _Count <= size(), "Count out of range in span::subspan()");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        return {data() + _Offset, _Count == dynamic_extent ? size() - _Offset : _Count};
+    }
+
+    constexpr span<element_type, dynamic_extent> subspan(index_type _Offset, index_type _Count = dynamic_extent) const
+        noexcept {
+#if _CONTAINER_DEBUG_LEVEL > 0
+        _STL_VERIFY(_Offset <= size(), "Offset out of range in span::subspan(offset, count)");
+        _STL_VERIFY(_Count == dynamic_extent || _Offset + _Count <= size(),
+            "Count out of range in span::subspan(offset, count)");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        return {data() + _Offset, _Count == dynamic_extent ? size() - _Offset : _Count};
+    }
+
+    // [span.obs], span observers
+    constexpr index_type size() const noexcept {
+        return _Mypair._Get_first().size();
+    }
+
+    constexpr index_type size_bytes() const noexcept {
+        return size() * static_cast<index_type>(sizeof(element_type));
+    }
+
+    _NODISCARD constexpr bool empty() const noexcept {
+        return size() == 0;

```

Comment:
Should we:
```c++
if constexpr (_Extent == dynamic_extent) {
  return size() == 0;
} else if constexpr (_Extent == 0) {
  return true;
} else {
  return false;
}
```
here for better debug codegen? Admittedly this feels like overkill even to me, and I *love* spamming `if constexpr` everywhere.

Reply:
Nah I would rather not